.Lib9c.Tests/ScenarioTests/CombinationAndRapidCombinationTest.cs

```diff
@@ -0,0 +1,269 @@
+            nextState = combinationEquipmentAction.Execute(new ActionContext
+            {
+                PreviousStates = nextState,
+                BlockIndex = 0,
+                Random = random,
+                Signer = _agentAddress,
+            });
+
+            var slot0Value = nextState.GetState(_slot0Address);
+            Assert.NotNull(slot0Value);
+            var slot0State = new CombinationSlotState((Dictionary)slot0Value);
+            Assert.NotNull(slot0State.Result.itemUsable);
+            var equipment = (Equipment)slot0State.Result.itemUsable;
+            var additionalStats = equipment.StatsMap
+                .GetAdditionalStats(true)
+                .ToArray();
+            var skills = equipment.Skills;
+            Assert.Equal(optionNumbers.Length, equipment.optionCountFromCombination);
+            var optionSheet = _tableSheets.EquipmentItemOptionSheet;
+            var mainAdditionalStatMin = 0;
+            var mainAdditionalStatMax = 0;
+            var requiredBlockIndex = 0;
+            foreach (var optionNumber in optionNumbers)
+            {
+                var optionInfo = subRecipeRow.Options[optionNumber - 1];
+                requiredBlockIndex += optionInfo.RequiredBlockIndex;
+                var optionRow = optionSheet[optionInfo.Id];
+                if (optionRow.StatMin > 0 || optionRow.StatMax > 0)
+                {
+                    if (optionRow.StatType == equipment.UniqueStatType)
+                    {
+                        mainAdditionalStatMin += optionRow.StatMin;
+                        mainAdditionalStatMax += optionRow.StatMax;
+                        continue;
+                    }
+
+                    var additionalStatValue = additionalStats
+                        .First(e => e.statType == optionRow.StatType)
+                        .additionalValue;
+                    Assert.True(additionalStatValue >= optionRow.StatMin &&
+                                additionalStatValue <= optionRow.StatMax + 1);
+                }
+                else if (optionRow.SkillId != default)
+                {
+                    var skill = skills.First(e => e.SkillRow.Id == optionRow.SkillId);
+                    Assert.True(skill.Chance >= optionRow.SkillChanceMin &&
+                                skill.Chance <= optionRow.SkillChanceMax + 1);
+                    Assert.True(skill.Power >= optionRow.SkillDamageMin &&
+                                skill.Power <= optionRow.SkillDamageMax + 1);
+                }
+            }
+
+            var mainAdditionalStatValue = additionalStats
+                .First(e => e.statType == equipment.UniqueStatType)
+                .additionalValue;
+            Assert.True(mainAdditionalStatValue >= mainAdditionalStatMin &&
+                        mainAdditionalStatValue <= mainAdditionalStatMax + 1);
+            Assert.Equal(requiredBlockIndex + 1, slot0State.RequiredBlockIndex);
+
+            // FIXME
```

Comment:
@longfin The tests after this line should be finished. However, since then the logic is being developed by different developers in different branches. I wrote a test beforehand, but it's failing.
I plan to move to another branch after this PR is merged and finish writing the tests.

Reply:
Then it would be better if we have comments about that here IMO.